src/NameService/NameService.cs

```diff
@@ -504,9 +505,10 @@ private static bool CheckIPv6(string ipv6)
                 {
                     if (fragment.Length > 4) return false;
                     int index = hasEmpty ? i + 8 - length : i;
-                    numbers[index] = (ushort)StdLib.Atoi(fragment, 16);
+                    numbers[index] = (ushort)(short)StdLib.Atoi(fragment, 16);
```

Comment:
I think that if is negative we should throw an exception

Reply:
`"ffff"` will be parsed to `-1`. So we need to convert it.